receiver/httpcheckreceiver/metadata.yaml

```diff
@@ -70,3 +70,38 @@ metrics:
       value_type: int
     unit: "s"
     attributes: [http.url, http.tls.issuer, http.tls.cn, http.tls.san]
+  httpcheck.dns_lookup.duration:
+    description: Time spent performing DNS lookup for the endpoint.
+    enabled: false
+    gauge:
+      value_type: int
+    unit: ms
+    attributes: [http.url]
+  httpcheck.tcp_connection.duration:
```

Comment:
i wonder if `http.client.connection.duration` (as per the semconv) is a better representation of what is being measured here. the protocol (tcp) is defined in the semantic conventions as an attribute rather than being part of the name

Reply:
This is where my lack of exposure to metrics has me a bit baffled. Should we have it where the "client.connection" is TCP so we just add an attribute saying "protocol = tcp" or would this be a deeper change where we want anything with a "p" suffix to be treated as a protocol. 

So like there would be 2 httpcheck.client.connection.duration metrics, one for protocol=tcp and one for protocol=http?

TLS and DNS _could_ be measured similarly, but TLS and DNS aren't measuring protocol-level connection durations.

Reply:
Also, it looks like the protocol attribute is named `network.transport` in some cases.